aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsProxyHttpServletRequest.java

```diff
@@ -437,23 +437,28 @@ public String getRemoteAddr() {
         if (request.getRequestContext() == null || request.getRequestContext().getIdentity() == null) {
             return "127.0.0.1";
         }
-        if (request.getRequestContext().getElb() != null) {
-            return request.getHeaders().get(CLIENT_IP_HEADER);
+        if (request.getRequestSource().equals(RequestSource.ALB)) {
+            return Objects.nonNull(request.getHeaders()) ?
+                    request.getHeaders().get(CLIENT_IP_HEADER) :
+                    request.getMultiValueHeaders().getFirst(CLIENT_IP_HEADER);
         }
         return request.getRequestContext().getIdentity().getSourceIp();
     }
 
 
     @Override
     public String getRemoteHost() {
-        if (Objects.nonNull(request.getRequestContext().getElb())) {
-            String hostHeader = request.getHeaders().get(HttpHeaders.HOST);
+        String hostHeader;
+        if (request.getRequestSource().equals(RequestSource.ALB)) {
+            hostHeader = Objects.nonNull(request.getHeaders()) ?
+                    request.getHeaders().get(HttpHeaders.HOST) :
+                    request.getMultiValueHeaders().getFirst(HttpHeaders.HOST);
 
             // the host header has the form host:port, so we split the string to get the host part
             return Arrays.asList(hostHeader.split(":")).get(0);
         }
-
-        return request.getMultiValueHeaders().getFirst(HttpHeaders.HOST);
+        hostHeader = request.getMultiValueHeaders().getFirst(HttpHeaders.HOST);
+        return Arrays.asList(hostHeader.split(":")).get(0);
```

Comment:
can you please remove the duplication, so we end up with a single return statement?